### DISCUSSION TITLE
Keyboard Azerty

### DISCUSSION THREAD

**Discussion by 15albedo** [Dakkaron/Fairberry#25] [Q&A]
Good morning,

First of all thank you for this project.

I just finished programming and assembling with an azerty keyboard (I'm French). However, I encounter a problem: 
- the alt key is inactive
- the sym key acts as if it were the alt key
- when I put the keyboard in qwerty in the phone settings, by pressing the sym key, I can correctly write the symbols # 1 2 3 ( ) _ - ...
but when the keyboard is in azerty in the phone settings, then by pressing the sym key, then all the symbols are mixed, to make a 9 I have to press the Y key for example.

Do you have an idea of ​​the lines in the code that I could modify so that azerty mode is functional and that the alt and sym keys work well?

Sorry for my English, it's a Google translation.

THANKS

**Comment by Dakkaron** [comment]
Cool that you are building one!

I'd recommend you to stay on US international layout on your phone and just adjust the keymaps in the keymaps.h file to suit your needs.

I had some trouble with keyboard layouts on Android before and the US international seems to do the trick best.

If that doesn't work, let me know and I'll add some layout config mechanism.

**Reply by 15albedo** [reply]
Good morning

Thank you for this response.

I set the keyboard to international English, and I reversed the position of the letters a z q and w in the keymaps.h file
It works flawlessly now.

However, I still have one problem:
It's the alt key that should activate the keyboard symbols, but it's always the sym key that does it.

Do you have an idea?

THANKS

**Reply by XantiaActiva** [reply]
> Cool that you are building one!
> 
> I'd recommend you to stay on US international layout on your phone and just adjust the keymaps in the keymaps.h file to suit your needs.
> 
> I had some trouble with keyboard layouts on Android before and the US international seems to do the trick best.
> 
> If that doesn't work, let me know and I'll add some layout config mechanism.

Hello Dakkaron,

thank you for your reply to @15albedo s topic, that makes it pretty clear to me. I got a keyboard with QWERTZ layout:

<img width="970" height="405" alt="grafik" src="https://github.com/user-attachments/assets/0e499462-d46a-47e5-b38e-5671fdbe475f" />

So I have to exchange the Z and the Y in the keymaps.h file, is that correct? Actually the layout is QWERTZ when using Germany, but all the other symbols on the keyboard are hard to find. If I set it to English US the layout is QWERTY, but I can access all other keys like printed onto the keyboard. So English US makes more sense (and I have to swap Y and Z). 

One more question: as @15albedo I have got the problem, that the sym key acts as the alt key. With pressed sym key I can access the 2nd level on the keyboard with its signs. When I press alt nothing happens in the most cases, some keys start actions if pressed together with alt. Is there a way to exchange the sim key and the alt key?

Thank you!

Best regards

Fred

**Reply by Dakkaron** [reply]
Hi,

Yeah, keep the keyboard layout as US international on the phone, change the keymaps.h to do what you want.

SYM is the second layer that's printed on the keys, so e.g. SYM+X = 8

Alt is forwared to the OS as the regular alt key, since it allows for some things inside the OS. Sym is not forwared and only used inside the keyboard to get to the second layer.

If you use US international layout you can e.g. use Alt+U for the "¨" dead key, so that Alt+U followed by O gives you "ö".

Other Alt combinations for US international:

- Alt+U followed by A/E/I/O/U: äëïöü
- Alt+U follewed by Shift + A/E/I/O/U: ÄËÏÖÜ
- Alt+I adds ˆ over the next character typed (like above)
- Alt+E adds ´ over the next character typed
- Alt+N: adds ˜ over the next character typed
- Alt+S: ß
- Alt+C: ç
- Some apps have key combinations with Alt. For example, Alt+D puts the focus on the address bar in Vivaldi Browser while Alt+F opens the menu.

So the Alt/Sym behavior is intentional and the same no matter which keyboard layout you use. Just make sure to activate US international layout.

On some phones (e.g. many Samsung phones) the US international layout is missing and can be installed via the app linked in the release: https://github.com/Dakkaron/Fairberry/releases/tag/v0.3.0 (US-International_External_Keyboard_Layout_base.apk)

**Reply by XantiaActiva** [reply]
Hello Dakkaron,

thank you! I exchanged the x and the z and now my keyboard is qwertz in the English US mode.

Best regards

Fred
